Sources/NIO/Codec.swift

```diff
@@ -674,3 +674,118 @@ extension MessageToByteHandler {
         }
     }
 }
+
+enum ByteToMessageDecoderVerificationError<OutputType>: Error {
+    case wrongDecodingResult(actual: OutputType, expected: OutputType)
+    case unexpectedDecodingResult(OutputType)
```

Comment:
This may want a rename to better distinguish `wrongDecodingResult` from `unexpectedDecodingResult`.

Reply:
done